pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -756,7 +756,7 @@ func (tc *testCase) runTestWithController(t *testing.T, hpaController *Horizonta
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	informerFactory.Start(ctx.Done())
-	go hpaController.Run(ctx)
+	go hpaController.Run(ctx, 5)
```

Comment:
IMHO we also need a test with multiple HPAs that would check this concurrent feature. Consider adding it to e2e https://github.com/kubernetes/kubernetes/tree/master/test/e2e/autoscaling

Reply:
Yes, can we please have a version of a test with, say, additional dummy 50 deployments+hpas so that we check that there are no concurrent writes to maps or other similar problems?

Reply:
Ack, agree, more testing is always good.

 I will try to add more tests asap, unfortunately I am quite busy at the moment. Adding e2e tests is imho not that simple, because I briefly checked the exisitng e2e tests and seems like they are not easily extensible in a way, that it can operate on multiple HPAs.